### ISSUE TITLE
Branding: custom colors

### ISSUE THREAD

**Issue by theoneandonly-vector** [Ylianst/MeshCentral#3578]
hey there

I know we can replace the images and Titles using config.json -settings.
But is there a way to change the blue-background with andother color / color-palette?

**Comment by JSuenram** [comment]
Hi,
this is something I am working on. But currently its only a prototype in my private fork/repository.
May take several weeks or even month until I can pull it.
Currently I am working on moving several missing styles to the CSS-File as there are many places where colors are still hard coded.

But there will be a full customization GUI sooner or later.... until then you need to change the style.css.
I believe there is a way to place it in another folder so it does not get overwritten on update....

**Comment by VINISHVV** [comment]
@JSuenram - Awaiting new GUI

**Comment by Ylianst** [comment]
As a temporary solution, next to `meshcentral-data`, you can create a folder called `meshcentral-web/public/styles' and copy the file 'node_modules/meshcentral/styles/style.css` into the new folder.

This will make it so that MeshCentral will use the `style.css` from the new folder instead of the default one. You can then change the style.css as you want with new colors, etc.

**Comment by JSuenram** [comment]
> As a temporary solution, next to `meshcentral-data`, you can create a folder called `meshcentral-web/public/styles' and copy the file 'node_modules/meshcentral/styles/style.css` into the new folder.

Yes this works really well, but be aware if there are new styles added to the main style.css you need to add them to the public/styles file also. But this is happening rarely....

**Comment by theoneandonly-vector** [comment]
a temporary "solution" to this would be to create a bash-script with sed-commands to copy the new file after each update and replace the colors.

> > As a temporary solution, next to `meshcentral-data`, you can create a folder called `meshcentral-web/public/styles' and copy the file 'node_modules/meshcentral/styles/style.css` into the new folder.
> 
> Yes this works really well, but be aware if there are new styles added to the main style.css you need to add them to the public/styles file also. But this is happening rarely....

**Comment by theoneandonly-vector** [comment]
does someone have a list of which colors need to be replaced?

**Comment by JSuenram** [comment]
If you open the Webpage and enter Developer-Mode in your Browser you can get information for the elements and styles.
![image](https://user-images.githubusercontent.com/68963439/152125971-22ec0263-59bb-446a-ac87-375ab6c4ed29.png)
![image](https://user-images.githubusercontent.com/68963439/152125995-4728940b-9807-4c1f-bc5a-de9fd1c2d98c.png)

Here you can see which style from style.css is used or if it is localy overwritten by hardcoded style in default.handlebars.
You can change any color which is already set in style.css. You can also change the hard coded colors in default-handlebars, but that gets overwritten on every update....

**Comment by theoneandonly-vector** [comment]
@JSuenram yes I know that, but maybe you alread yhave a list of all the values so I don't need to search :)

**Comment by JSuenram** [comment]
> do I need to edit another file to change mastheads background-color on /terms? smh that is not changing when chaning the color in style.css
In webserver.js there is the rendering for Termns of Service which can also come from a terms.txt when placed in there in meshcentral-data

Beginngin on Line 3090. It looks like the terms can come from Database and are translated. 

But this is only "guessed", have not investigated that further....

**Comment by JSuenram** [comment]
Just a small peak, making some progress. Getting first ideas how all this bunch of super exciting code works....
What a genius little piece of software.... 👍 

![image](https://user-images.githubusercontent.com/68963439/152356156-32e8e5be-1717-49bf-a65b-4927837efde5.png)

**Comment by VINISHVV** [comment]
@JSuenram - Great i assume the new customization tab is restricted for Admin users only.

**Comment by JSuenram** [comment]
> @JSuenram - Great!!!
> 
> i assume the new customization tab is restricted for Admin users only.

Yes, only admins will be allowed to do this.

**Comment by ne0YT** [comment]
> > @JSuenram - Great!!!
> > i assume the new customization tab is restricted for Admin users only.
> 
> Yes, only admins will be allowed to do this.

Hey there

I have changed quite everything I think now using the style.css, but I cannot find where to change what to recolor this blue bar, when sharing a connection using a link, maybe you know how to do this currently?:
![image](https://user-images.githubusercontent.com/97364807/225650337-8299e73d-ea42-4987-8396-2acc7e42cf89.png)

**Comment by silversword411** [comment]
> I have changed quite everything I think now using the style.css, 

Where is the repo with those tweaks?

**Comment by ne0YT** [comment]
> > I have changed quite everything I think now using the style.css,
> 
> Where is the repo with those tweaks?

there is no repo with those tweaks
